# Generated by Django 2.2.11 on 2021-04-21 05:58

import django.contrib.gis.db.models.fields
from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ('tracking', '0009_device_deleted'),
    ]

    operations = [
        migrations.CreateModel(
            name='InvalidLoggedPoint',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('point', django.contrib.gis.db.models.fields.PointField(editable=False, null=True, srid=4326)),
                ('heading', models.PositiveIntegerField(default=0, editable=False, help_text='Heading in degrees')),
                ('velocity', models.PositiveIntegerField(default=0, editable=False, help_text='Speed in metres/hr')),
                ('altitude', models.IntegerField(default=0, editable=False, help_text='Altitude above sea level in metres')),
                ('seen', models.DateTimeField(editable=False, null=True)),
                ('source_device_type', models.CharField(choices=[('tracplus', 'TracPlus'), ('iriditrak', 'Iriditrak'), ('dplus', 'DPlus'), ('spot', 'Spot'), ('dfes', 'DFES'), ('mp70', 'MP70'), ('fleetcare', 'fleetcare'), ('other', 'Other'), ('fleetcare_error', 'Fleetcare (error)')], default='other', max_length=32)),
                ('deviceid', models.CharField(db_index=True, max_length=32, null=True)),
                ('device_id', models.IntegerField(db_index=True, null=True)),
                ('raw', models.TextField(editable=False)),
                ('category', models.CharField(choices=[(1, 'Invalid Raw Data'), (10, 'Invalid Timestamp'), (11, 'Invalid Timestamp Format'), (20, 'Future Data')], max_length=32)),
                ('message', models.TextField()),
                ('created', models.DateTimeField(auto_now_add=True, db_index=True)),
            ],
            options={
                'ordering': ['-seen'],
                'abstract': False,
            },
        ),
    ]
